components/script/dom/cssstyledeclaration.rs

```diff
@@ -305,31 +305,52 @@ impl CSSStyleDeclaration {
         DOMString::from(string)
     }
 
-    fn set_property(
+    /// <https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-setproperty>
+    ///
+    /// This function receives a `PotentiallyParsedPropertyId` instead of a `DOMString` in case
+    /// the caller already has a parsed property ID.
+    fn set_property<P>(
         &self,
-        id: PropertyId,
+        id: P,
         value: DOMString,
         priority: DOMString,
         can_gc: CanGc,
-    ) -> ErrorResult {
-        // Step 1
+    ) -> ErrorResult
+    where
+        PotentiallyParsedPropertyId: From<P>,
+    {
+        // Step 1. If the readonly flag is set, then throw a NoModificationAllowedError exception.
```

Comment:
This signature is generic because stylo relies on it via the [`css_properties_accessors`](https://doc.servo.org/style/macro.css_properties_accessors.html) macro.

Reply:
Oh right, thats better.
 sidenote: I wonder how if the compiler is able to deduplicate the closure in the generic function, since it doesn't depend on the generic parameter.